src/input.cpp

```diff
@@ -1145,38 +1157,205 @@ std::string input_context::get_desc(
+    if( n_text > 0 ) {
+        padded_text = string_format( "\u00A0%s", padded_text );
+    }
     if( na ) {
         //~ keybinding description for unbound or non-applicable keys
-        return string_format( separate_fmt, pgettext( "keybinding", "n/a" ), text );
-    } else {
-        return string_format( separate_fmt, get_desc( action_descriptor, 1, evt_filter ), text );
+        return string_format( separate_fmt, pgettext( "keybinding", "n/a" ), padded_text );
     }
+    return string_format( separate_fmt, get_desc( action_descriptor, 1, evt_filter ), padded_text );
 }
 
 std::string input_context::get_desc(
     const std::string &action_descriptor,
     const std::string &text,
-    const input_event_filter &evt_filter ) const
+    const keybinding_hint_state state,
+    const bool show_separators,
+    const input_event_filter &evt_filter
+) const
+{
+    translation inline_fmt;
+    translation separate_fmt;
+    std::string lb_sep;
+    std::string rb_sep;
+    std::string lp_sep;
+    std::string rp_sep;
+    if( show_separators ) {
+        lb_sep = get_option<std::string>( "KEYBINDING_HINT_SEPARATE_SEPARATOR_LEFT" );
+        rb_sep = get_option<std::string>( "KEYBINDING_HINT_SEPARATE_SEPARATOR_RIGHT" );
+        lp_sep = get_option<std::string>( "KEYBINDING_HINT_INLINE_SEPARATOR_LEFT" );
+        rp_sep = get_option<std::string>( "KEYBINDING_HINT_INLINE_SEPARATOR_RIGHT" );
+    }
+    std::string c_key = get_tag_from_color( get_hint_color_for_key( state ) );
+    if( state != keybinding_hint_state::NONE ) {
+        std::string c_sep = get_tag_from_color( get_hint_color_for_separator( state ) );
+        std::string c = get_tag_from_color( get_hint_color( state ) );
+        inline_fmt = to_translation( "keybinding",
+                                     string_format( "%s%s</color>%s%s</color>%s%s</color>%s%s</color>%s%s</color>", c, "%1$s", c_sep,
+                                             lp_sep, c_key, "%2$s", c_sep, rp_sep, c, "%3$s" ) );
+        separate_fmt = to_translation( "keybinding",
+                                       string_format( "%s%s</color>%s%s</color>%s%s</color>%s%s</color>", c_sep, lb_sep, c_key, "%1$s",
+                                               c_sep, rb_sep, c, "%2$s" ) );
+    } else {
+        //  %1$s: action description text before key,
+        //~ %2$s: key description,
+        //~ %3$s: action description text after key.
+        inline_fmt = to_translation( "keybinding", string_format( "%s%s%s%s</color>%s%s", "%1$s", lp_sep,
+                                     c_key,
+                                     "%2$s", rp_sep, "%3$s" ) );
+        // \u00A0 is the non-breaking space
+        //~ %1$s: key description,
+        //~ %2$s: action description.
+        separate_fmt = to_translation( "keybinding", string_format( "%s%s%s</color>%s%s", lb_sep, c_key,
+                                       "%1$s",
+                                       rb_sep, "%2$s" ) );
```

Comment:
Same here. Also the comments starting with "//~" are intended for translators and only apply to the extracted format strings, so the translators will not know what "%1$s" etc refer to after this change.